sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -465,6 +465,8 @@ object SimplifyConditionals extends Rule[LogicalPlan] with PredicateHelper {
         if cond.deterministic && trueValue.semanticEquals(falseValue) => trueValue
       case If(cond, l @ Literal(null, _), FalseLiteral) if !cond.nullable => And(cond, l)
       case If(cond, l @ Literal(null, _), TrueLiteral) if !cond.nullable => Or(Not(cond), l)
+      case If(cond, FalseLiteral, l @ Literal(null, _)) if !cond.nullable => And(Not(cond), l)
+      case If(cond, TrueLiteral, l @ Literal(null, _)) if !cond.nullable => Or(cond, l)
```

Comment:
minor, can we call `l` as `nullLiteral`?

```
      case If(cond, nullLiteral @ Literal(null, _), FalseLiteral) if !cond.nullable => And(cond, nullLiteral)
      case If(cond, nullLiteral @ Literal(null, _), TrueLiteral) if !cond.nullable => Or(Not(cond), nullLiteral)
      case If(cond, FalseLiteral, nullLiteral @ Literal(null, _)) if !cond.nullable => And(Not(cond), nullLiteral)
      case If(cond, TrueLiteral, nullLiteral @ Literal(null, _)) if !cond.nullable => Or(cond, nullLiteral)
```

Reply:
Hmm that will break each of these into two lines which may affect readability, also I think given these are already `Literal(null, _)`, naming the variable to `nullLiteral` doesn't add much value maybe?